code/modules/vehicles/tonk.dm

```diff
@@ -0,0 +1,354 @@
+	ammo.current_rounds--
+	return TRUE
+
+/obj/item/weapon/tank_weapon/proc/can_fire(var/turf/T)
+	if(T in orange(owner, 2))
+		if(owner.gunner)
+			to_chat(owner.gunner, "Firing your main gun here could damage the tank!")
+			return FALSE
+	return TRUE
+
+/obj/item/weapon/tank_weapon/minigun/can_fire(var/turf/T)
+	return TRUE //No loc check here
+
+/obj/vehicle/tonk
+	name = "MK-1 Benocrusher x10"
+	desc = "A gigantic wall of metal designed for maximum beno destruction. Click it with an open hand to enter as a pilot or a gunner."
+	icon = 'icons/obj/tonk.dmi'
+	icon_state = "tank"
+	layer = 3
+	bound_width = 128
+	bound_height = 128
+	anchored = FALSE
+	can_buckle = FALSE
+	req_access = list(ACCESS_MARINE_TANK)
+	move_delay = 2
+	//Who's driving the tonk
+	var/mob/living/carbon/human/pilot
+	var/mob/living/carbon/human/gunner
+	var/list/operators = list() //Who's in this tank? Prevents you from entering the tank again
+	//Health and combat shit
+	var/obj/turret_overlay/turret_overlay //Allows for independantly swivelling guns, wow!
+	var/obj/turret_overlay/minigun/minigun_overlay
+	var/max_health = 300 //i dunno what a good number for this is, change this maintainers!
+	var/obj/item/weapon/tank_weapon/main_cannon //What we use to shoot big shells
+	var/obj/item/weapon/tank_weapon/minigun/minigun //What we use to shoot mini shells ((rapidfire xenocrusher 6000))
+	var/main_cannon_dir = null //So that the guns swivel independantly
+	var/minigun_dir = null
+	var/firing = FALSE //Used in autofire
+	var/turf/stored_firing_turf = null
+
+/obj/turret_overlay
+	name = "Tank gun turret"
+	desc = "Big fucking tonk gun oh lawd he comin"
+	icon = 'icons/obj/tonk_gun.dmi'
+	icon_state = "turret"
+	layer = 3.1
+	animate_movement = TRUE //So it doesnt just ping back and forth and look all stupid
+	mouse_opacity = FALSE //It's an overlay
+
+/obj/vehicle/tonk/Initialize()
+	. = ..()
+	turret_overlay = new(get_turf(src))
+	update_icon()
+	health = max_health
+	main_cannon = new(src) //Make our guns
+	minigun = new(src)
+	main_cannon.owner = src
+	minigun.owner = src
+
+/obj/vehicle/tonk/Destroy()
+	qdel(turret_overlay)
+	qdel(main_cannon)
+	qdel(minigun)
+	. = ..()
+
+/obj/vehicle/tonk/Move()
+	. = ..()
+	update_icon()
+
+/obj/vehicle/tonk/update_icon() //To show damage, gun firing, whatever. We need to re apply the gun turret overlay.
+	var/icon/I = icon(icon,icon_state,dir)
+	bound_width = I.Width() //Adjust the hitbox in case admins want to make an OMEGATONK
+	bound_height = I.Height()
+	if(!turret_overlay) //We have to do this because of BYOND's innate dir setting habits, in byond, if you have an overlay'd object its direction can ONLY be which way the source is facing. So, we improvise.. ((yes this is kinda shit))
+		return
+	if(main_cannon_dir)
+		turret_overlay.setDir(main_cannon_dir)
+		if(main_cannon_dir == WEST || main_cannon_dir == SOUTHWEST|| main_cannon_dir == NORTHWEST)
+			turret_overlay.pixel_x = -19
```

Comment:
I dont know what brittle code is but this is the only way i can make the gun swivel properly due to how BYOND handles overlays

Reply:
You shouldn't be setting the offset like this, handle it in a generic fashion for turning or make the overlay art itself handle it

Reply:
Just so you know, @Kmc , the existing tank never rotated its turret on firing. Since the 3/4 perspective makes rotating the turret require one sprite for each  direction, consider this not a really necessary feature if it makes your job harder.
The way the current tank handles it is that you can only fire the main gun in the direction the tank is facing, which forces coordination between gunner and driver.
Not saying changing this is bad, merely for you to consider the cost and if it's worth the effort for you.
Though support for the functionality wouldn't be bad, for if we add a different gun whose sprite allows for smoother 360 rotation.